gigascience/gigadb-website
# Improve Accessibility for Conditional Form Inputs Based on "Image Status"

# Pull request for issue: #1471

This is a pull request for the following functionalities:

* Improved accessibility related to image metafields, e.g.
  * http://gigadb.gigasciencejournal.com:9170/adminDataset/create
  * http://gigadb.gigasciencejournal.com:9170/adminDataset/update/id/5

## How to test?

* Navigate to aboves pages
* Interact with the image input
* Check that screen reader (voiceover on mac, NVDA on windows) announces something related to the metafields being shown / hidden --note: this is a bit easier to test on chrome as safari doesn't allow to clear an input of type field
* When tabbing through image metafields, the image metafields label is announced

## How have functionalities been implemented?

* Added `aria-controls` to the file input to programmatically define a relationship with image metafields
* Toggle aria-hidden for image metafields
* wrap metafields in a fieldset and legend so that they are grouped. This helps screen reader users identify them
* Added live region that announces image metafields being toggled

## Any issues with implementation?

None

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
